### PR TITLE
Disable implicit casts

### PR DESCRIPTION
**Pull request by isoos**
Going ahead of these changes...

### CODE REVIEW

File: lib/src/license.dart [hunk 1]
```diff
@@ -17,15 +17,16 @@ Future<List<LicenseFile>> detectLicensesInDir(String baseDir) async {
   final rootFiles = await new Directory(baseDir).list().toList();
   final licenseCandidates = rootFiles
       .where((fse) => fse is File)
+      .cast<File>()
```
Comment (jakobr-google) [inline lib/src/license.dart]: I believe the recommendation is to replace these two lines with a single:
```dart
  .whereType<File>()
```

Comment (isoos) [reply lib/src/license.dart]: I've changed it, but getting

````
  UnimplementedError: whereType is not yet supported
  dart:collection/list.dart 198:7               _ListBase&Object&ListMixin.whereType
  package:pana/src/license.dart 19:8            detectLicensesInDir
````

Changing it back for now.

File: lib/src/model.g.dart [hunk 2]
```diff
@@ -15,7 +15,7 @@ Summary _$SummaryFromJson(Map<String, dynamic> json) => new Summary(
     json['packageName'] as String,
     json['packageVersion'] == null
         ? null
-        : new Version.parse(json['packageVersion']),
+        : new Version.parse(json['packageVersion'] as String),
```
Comment (kevmoo) [inline lib/src/model.g.dart]: Did you make these changes by hand?

Comment (isoos) [reply lib/src/model.g.dart]: Updated the `tool/src/version_helper.dart`

**Comment by kevmoo** [comment]
Ah!

**Comment by isoos** [comment]
Interesting thing: the PR is passes test right now, but if I run it with preview-dart-2, the end-to-end test fail:

````
 find test -name \*_test.dart -exec dart --preview-dart-2 {} \;
````

The error message is: `Which: has different length and is missing map key 'constraint' at location ['pkgResolution']['dependencies'][1]`, which would indicate that it fails to serialize the `VersionConstraint` instance. @kevmoo: any idea what could be wrong there?

**Comment by kevmoo** [comment]
No clue. Very weird!

File: lib/src/license.dart (hunk 1 above)
Comment (kevmoo) [reply lib/src/license.dart]: Need to wait until preview-dart-2 is enabled everywhere...

**Comment by isoos** [comment]
Found the issue: `Map<String, dynamic> get dependencies` in `Pubspec` was returning null because of the type check failing. Using wrapping solved it for now, added TODO to check back on it later.

**Pull request merged**